src/services/htmlCompletion.ts

```diff
@@ -175,9 +175,41 @@ export function doComplete(document: TextDocument, position: Position, htmlDocum
 				});
 			});
 		});
+		collectDataAttributesSuggestions(range);
 		return result;
 	}
 
+	function collectDataAttributesSuggestions(range: Range) {
+
+		const dataAttr = 'data-';
+		let dataAttributes: string[] = [];
+
+		function addNodeDataAttributes(node: Node) {
+			node.attributeNames.filter(attr => attr.substr(0, dataAttr.length) === dataAttr).forEach(attr => {
```

Comment:
I'd suggest doing a single forEach loop (no `filter`)
use `startsWith` from `../utils/strings` for the check.

Reply:
thanks for the review 👍